tasks/Task.cpp

```diff
@@ -28,8 +28,9 @@ bool Task::configureHook()
         return false;
 
     // Creating the motion model object
-    delete model_simulation;
-    model_simulation = new ModelSimulation(simulator, TaskContext::getPeriod(), _sim_per_cycle.get(), 0);
+    std::auto_ptr<ModelSimulation> sim(new ModelSimulation(simulator, TaskContext::getPeriod(), _sim_per_cycle.get(), 0));
+    model_simulation = sim.release();
```

Comment:
Used like this, it makes no difference. If the constructor raises, then the object is never initialized. If the constructor passes, then you release it right away.

If you move the release just before the `return true;`, then things are becoming more interesting. If the parameters are invalid (in `setUWParameters`) and you raise, then you keep your invariants.

If you get a crash, run under valgrind. It should give you a better idea of what's going on.

Reply:
what was going on was that he was deleting the pointer twice (cleanupHook AND configureHook)

Reply:
But .. the PR was adding the reinit to NULL. I'm confused. Anyways, glad I could help ;-)

Reply:
Thanks guy.
I tested it today and had no problem.
Merging it